src/Raven.Server/ServerWide/CompareExchangeExpirationStorage.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using Raven.Client;
+using Raven.Client.Util;
+using Raven.Server.Rachis;
+using Raven.Server.ServerWide.Context;
+using Sparrow;
+using Sparrow.Binary;
+using Sparrow.Json;
+using Voron;
+using Voron.Data.Tables;
+
+namespace Raven.Server.ServerWide
+{
+    public class CompareExchangeExpirationStorage
+    {
+        public static string CompareExchangeByExpiration = "CompareExchangeByExpiration";
+        private static Dictionary<Slice, List<Slice>> _expired;
```

Comment:
Make this an instance class, and then place an instance of the cluster state machine.
In this way, you cannot use it on multiple raft engines at the same time, and we need that, for testing, if nothing else.